var/spack/repos/builtin/packages/pastix/package.py

```diff
@@ -0,0 +1,89 @@
+    maintainers = ['fpruvost', 'mfaverge', 'ramet']
+
+    version('master', branch='master', submodules=True)
+    version('6.2.1', '50742cc0e6e03728f7529fc607d5c65b9e14205f192947678d6103a872a6210c')
+
+    # cmake's specific
+    variant('shared', default=True, description='Build Pastix as a shared library')
+
+    # pastix's specific
+    variant('doc', default=False, description='Enable documentation')
+    variant('int64', default=False, description='To use 64 bits integers')
+    variant('metis', default=False, description='Enable Metis')
+    variant('scotch', default=True, description='Enable Scotch')
+    variant(
+        'runtime', default='none', description='Runtime support',
+        values=('none', 'starpu'), multi=False
+    )
+    variant('cuda', default=False, when='runtime=starpu', description='Enable CUDA')
+    variant('mpi', default=False, description='Enable MPI')
+
+    # Dependencies
+    depends_on("pkg-config", type='build')
+    depends_on("hwloc")
+    depends_on("lapack")
+    with when("+metis"):
+        depends_on("metis@5.1:")
+        depends_on("metis@5.1:+int64", when='+int64')
+    with when("+scotch"):
+        depends_on("scotch~metis")
+        depends_on("scotch~metis+int64", when='+int64')
+        depends_on("scotch~metis~mpi", when='~mpi')
+    with when("runtime=starpu"):
+        depends_on("starpu")
+        depends_on("starpu~mpi", when='~mpi')
+        depends_on("starpu+mpi", when='+mpi')
+        depends_on("starpu~cuda", when='~cuda')
+        depends_on("starpu+cuda", when='+cuda')
+        depends_on("cuda", when='+cuda')
+    depends_on("mpi", when='+mpi')
+
+    def cmake_args(self):
+        spec = self.spec
+
+        args = [
+            "-Wno-dev",
+            self.define("CMAKE_COLOR_MAKEFILE", "ON"),
+            self.define("CMAKE_VERBOSE_MAKEFILE", "ON"),
+            self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
+            self.define_from_variant("PASTIX_INT64", "int64"),
+            self.define_from_variant("PASTIX_ORDERING_METIS", "metis"),
+            self.define_from_variant("PASTIX_ORDERING_SCOTCH", "scotch"),
+            self.define("PASTIX_WITH_PARSEC", "OFF"),
+            self.define_from_variant("PASTIX_WITH_MPI", "mpi")
+        ]
+
+        if spec.satisfies('runtime=starpu'):
+            args.extend([self.define("PASTIX_WITH_STARPU", "ON")])
+            args.extend([self.define_from_variant("PASTIX_WITH_CUDA", "cuda")])
+
+        if ('^intel-mkl' in spec or '^intel-parallel-studio+mkl' in spec):
+            if ('threads=none' in spec):
+                args.extend([self.define("BLA_VENDOR", "Intel10_64lp_seq")])
+            else:
+                args.extend([self.define("BLA_VENDOR", "Intel10_64lp")])
+        elif ('^netlib-lapack' in spec):
+            args.extend([self.define("BLA_VENDOR", "Generic")])
+        elif ('^openblas' in spec):
+            args.extend([self.define("BLA_VENDOR", "OpenBLAS")])
```

Comment:
Here again, OpenBLAS has to be built with a sequential version.

Reply:
This is the case by default in spack, openblas is sequential (variant threads=none). It is not possible to force it at the cmake  level because the library filename is the same (seq. vs. multithreaded).

Reply:
In fact threads=None is not enough for OpenBLAS as the lib is not reentrant. 
If you look at the packages from Debian/ubuntu for example, what is important is the fact that OpenBLAS has been built with `USE_OPENMP=1`, but I don't know how to check if it's the case or not.

Reply:
I have added the following to fix this `depends_on('openblas threads=openmp', when='^openblas')`